pkg/operator/controller/ingress_status.go

```diff
@@ -101,3 +126,156 @@ func ingressStatusesEqual(a, b operatorv1.IngressControllerStatus) bool {
 
 	return true
 }
+
+func isProvisioned(service *corev1.Service) bool {
+	ingresses := service.Status.LoadBalancer.Ingress
+	return len(ingresses) > 0 && (len(ingresses[0].Hostname) > 0 || len(ingresses[0].IP) > 0)
+}
+
+func isPending(service *corev1.Service) bool {
+	return !isProvisioned(service)
+}
+
+func getServiceOwnerIfMatches(service *corev1.Service, matches func(*corev1.Service) bool) []string {
+	if !matches(service) {
+		return []string{}
+	}
+	controller, ok := service.Labels[manifests.OwningIngressControllerLabel]
+	if !ok {
+		return []string{}
+	}
+	return []string{controller}
+}
+
+func indexLoadBalancerControllerByName(obj runtime.Object) []string {
```

Comment:
I'm not sure about the name; what do you think of `indexControllerWithLoadBalancerByName`?

Reply:
better, but can we do it in a followup so this has some time to soak?